xarray/tests/test_backends_datatree.py

```diff
@@ -62,6 +63,52 @@ def test_netcdf_encoding(self, tmpdir, simple_datatree):
         with pytest.raises(ValueError, match="unexpected encoding group.*"):
             original_dt.to_netcdf(filepath, encoding=enc, engine=self.engine)
 
+    def test_open_datatree(self):
+        """Test `open_datatree` with netCDF4 file with this structure:
+        DataTree('None', parent=None)
+        │   Dimensions:        (lat: 1, lon: 2)
+        │   Dimensions without coordinates: lat, lon
+        │   Data variables:
+        │       root_variable  (lat, lon) float64 16B ...
+        └── DataTree('Group1')
+                Dimensions:      (lat: 2, lon: 2)
+                Dimensions without coordinates: lat, lon
+                Data variables:
+                    group_1_var  (lat, lon) float64 32B ...
+        """
+        filepath = os.path.join(
+            os.path.dirname(__file__), "data", "test_data_not_aligned.nc"
+        )
+        with pytest.raises(ValueError):
+            open_datatree(filepath)
+
+    def test_open_groups(self):
```

Comment:
I added `"test_data_not_aligned.nc"` to the `tests/data` directory. I could definitely add more test files and tests but didn't want to bloat the directory with too many test data files.